### PR TITLE
Final Project

### PR DESCRIPTION
**Pull request by sneeu-leeu**
## The Objectives of this projects:
- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write unit tests with React Testing Library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply, React best practices and language style guides in code.
- Use store, actions, and reducers in React.
- Perform a code review for a team member

### CODE REVIEW

File: src/redux/missions/missions.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const missions = await fetchAllMissions();
+  missions.forEach((mission) => {
+    const newMission = {
+      mission_id: mission.mission_id,
+      mission_name: mission.mission_name,
+      description: mission.description,
+    };
+    dispatch(addMission(newMission));
+  });
+};
```
Comment (vichuge) [inline src/redux/missions/missions.js]: - [x] When you take the missions info at this point, you add every mission one by one user and action, a better way to do it is to add all of them in an object and once you're here on the reducer, add it with another switch option, that would save the memory cost to run an action a lot of times. Please use this on this redux. 🐶

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your PR messages are descriptive ✔️

## Aspects to improve ♻️

- [x] I'm afraid I couldn't find your video presentation, I tried on your pull request description and readme file but I couldn't find it. Please, add the link for this (remember the video should follow all the requirements).

- [ ] It looks like your project doesn't have tests, please, I kindly invite you to add tests following the requirements.
image


- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sneeu-leeu** [comment]
Hi @vichuge, Thank you for taking the time to review our project! 

Could you please expand on your review regarding the tests? Our Project requirements only state to set up a testing environment. We have no required tests to write for this project on our documentation. What tests are you referring to?

To be more specific, the only reference to testing is:

"Technical set up**
 - Set up the repository on GitHub and use Gitflow.
 - Set up React and Redux.
 - Set up testing libraries.
 - Do not preserve user data."

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi **@sneeu-leeu**  and **@uwadonat** , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. :muscle:

**Changes Required**
Regarding the unit tests as pointed by the previous Code Reviewer, please refer to your learning objectives. To be precise, `bullet number 6`. You can find it [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md).


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

**Comment by sneeu-leeu** [comment]
@oxenprogrammer Thanks for the quick review. I am not going to ask for a second opinion purely for time reasons, but please read the entire document, not just the bullet points. There are NO TESTS required by Microverse for this project.

File: src/App.test.js [hunk 2]
```diff
@@ -3,6 +3,6 @@ import App from './App';
 
 test('renders learn react link', () => {
   render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
+  const linkElement = screen.getByText(/Missions/i);
   expect(linkElement).toBeInTheDocument();
 });
```
Comment (MahmoudBakr23) [inline src/App.test.js]: - [ ] Please add unit tests using the mentioned test libraries that you set up in your app 👍

**Comment by MahmoudBakr23** [comment]
> @oxenprogrammer Thanks for the quick review. I am not going to ask for a second opinion purely for time reasons, but please read the entire document, not just the bullet points. There are NO TESTS required by Microverse for this project.

Hi team 😄 
Please note that as long as it is mentioned in the bullet points within the project requirements, then it's mandatory 🙏 Yet, tests are very important to work with, it helps you to understand the technologies you're working with more deeply and also, it makes you see your code from a different perspective 🙂

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi team😃, 

Great job so far building this app and working on the requested changes. However, there's just a small thing left to do in order to get the approval 🙏 😄 

## Changes Required

_Please see the comment down below._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sneeu-leeu** [comment]
@MahmoudBakr23  Thanks for your review. After reaching out to student success, we confirmed with Anne-Lore that this project does not need tests. Next reviewer, please read the entire document

**Comment by SpaYco** [comment]
@sneeu-leeu please check:

https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#learning-objectives

> Write unit tests with React Testing Library.

https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#technical-set-up

> Set up testing libraries.

**Comment by Davidosky007** [comment]
## Hi @sneeu-leeu 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
* Kindly add unit tests to your project as it is a requirement [Check senior T.S.E response](https://github.com/sneeu-leeu/space-travelers-hub/pull/37#issuecomment-917417129l)
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Emmyn5600 — APPROVED** [review]
Hi Team,

Good job so far! 👏🏼 @sneeu-leeu  and @uwadonat

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- The project looks awesome:heavy_check_mark:
- Tests are passing successfully:heavy_check_mark: 
cpp18

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewers.

#### Optional suggestions
 
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**